src/app/ReadClient.cpp

```diff
@@ -883,7 +883,7 @@ CHIP_ERROR ReadClient::ComputeLivenessCheckTimerTimeout(System::Clock::Timeout *
     // TODO: We need to find a good home for this logic that will correctly compute this based on transport. For now, this will
     // suffice since we don't use TCP as a transport currently and subscriptions over BLE aren't really a thing.
     //
-    const auto & ourMrpConfig = GetDefaultMRPConfig();
+    auto ourMrpConfig = GetLocalMRPConfig().ValueOr(GetDefaultMRPConfig());
```

Comment:
Is the return value of ValueOr() guaranteed to survive long enough to be copied into ourMrpConfig?

It would be safer to hold a reference to the return value of GetLocalMRPConfig()...

Reply:
updated, thanks